### PR TITLE
Document health, metrics & OpenTracing extensions

### PR DESCRIPTION
**Pull request by jamesnetherton**
I wrote some docs for extensions that I added. 

I tweaked the doc extensions list mojo so that If you drop an adoc file into the docs extensions directory, it'll be linked to in the extensions summary readme. 

It'd be nice to eventually template & auto generate this stuff. For now, I just want to get an initial baseline added.

### CODE REVIEW

File: docs/modules/ROOT/pages/extensions/microprofile-health.adoc [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+The MicroProfile Health extension bridges the Camel health check API with MicroProfile Metrics & SmallRye Health.
+
+Maven users will need to add the following dependency to their `pom.xml` for this extension.
+
+[source,xml]
+------------------------------------------------------------
+<dependency>
+    <groupId>org.apache.camel.quarkus</groupId>
+    <artifactId>camel-quarkus-microprofile-health</artifactId>
+    <version>${camel-quarkus.version}</version>
```
Comment (ppalaga) [inline docs/modules/ROOT/pages/extensions/microprofile-health.adoc]: Aren't we (or shouldn't we if we don't) generally encouraging the users to import our BOM? So maybe we could omit the version here.

File: docs/modules/ROOT/pages/extensions/microprofile-health.adoc [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+</dependency>
+------------------------------------------------------------
+
+== Usage
+
+You can register health checks for your applications with the https://camel.apache.org/manual/latest/health-check.html[Camel health check API].
```
Comment (ppalaga) [inline docs/modules/ROOT/pages/extensions/microprofile-health.adoc]: @zregvart Is this the preferred way to link documents from other Camel guides? I know there is `xref:` - is that one not better for some reason?

File: docs/modules/ROOT/pages/extensions/microprofile-health.adoc (hunk 1 above)
Comment (jamesnetherton) [reply docs/modules/ROOT/pages/extensions/microprofile-health.adoc]: Yeah, I did ponder whether to bother including the version. I'm happy to remove it.

File: docs/modules/ROOT/pages/extensions/microprofile-health.adoc (hunk 2 above)
Comment (zregvart) [reply docs/modules/ROOT/pages/extensions/microprofile-health.adoc]: `xref` would be preferred, with `xref` there's the option of pointing to a specific version of the document and it's not tied to the URL structure of the website.

Comment (ppalaga) [reply docs/modules/ROOT/pages/extensions/microprofile-health.adoc]: Would the `antora/xref-validator` we run in this repo pass even if the health-check document is in another git repository?

**Pull request merged**